src/txgraph.cpp

```diff
@@ -1880,7 +1882,7 @@ TxGraph::GraphIndex TxGraphImpl::CountDistinctClusters(std::span<const Ref* cons
     std::vector<Cluster*> clusters;
     clusters.reserve(refs.size());
     for (const Ref* ref : refs) {
-        if (ref == nullptr) continue;
+        Assume(ref);
```

Comment:
Do you want to retain the safe exit if this isn't true? `if (!Assume(ref)) continue;` ?

Reply:
Current thinking and elsewhere in TxGraph we don't take the cost of a conditional if we're confident that the code cannot be reached.

See https://github.com/bitcoin/bitcoin/pull/31363#discussion_r1945148422 , https://github.com/bitcoin/bitcoin/pull/31363#discussion_r1945149772 , and https://github.com/bitcoin/bitcoin/pull/32100#issuecomment-2740685232 for more discussion